From 5f33436c55f6f905677005fdf2d917e0d1a57a60 Mon Sep 17 00:00:00 2001
From: Muhammad Aksar <muhammad.aksar@intel.com>
Date: Fri, 6 Dec 2019 18:36:57 +0530
Subject: [PATCH] Camera: Fix app crash while Connecting USB Camera

 -Camera app is crashing due to sending an error
  notification to the service while unsuccessful
  enumeration of the Camera device
 -Then need to reboot the system forcefully
 -Can be avoided this crash just by avoid sending
  error notification and instead can be breaking
  from that case and which handles that situation
 -It will not make any issue in the normal Camera
  use cases.

Tracked-On: OAM-88936
Signed-off-by: Muhammad Aksar <muhammad.aksar@intel.com>
---
 services/camera/libcameraservice/CameraService.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index ec83193..6dcce89 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -1917,7 +1917,7 @@ Status CameraService::supportsCameraApi(const String16& cameraId, int apiVersion
         case -1: {
             String8 msg = String8::format("Unknown camera ID %s", id.string());
             ALOGE("%s: %s", __FUNCTION__, msg.string());
-            return STATUS_ERROR(ERROR_ILLEGAL_ARGUMENT, msg.string());
+            break;
         }
         default: {
             String8 msg = String8::format("Unknown device version %x for device %s",
-- 
2.7.4

